src/parser/metrics/RealLinesOfCode.ts

```diff
@@ -45,7 +45,7 @@ export class RealLinesOfCode implements Metric {
             realLinesOfCode.add(currentNode.startPosition.row);
 
             // This is a leaf node, so add further lines if it spans over multiple lines and is no line ending:
-            if (currentNode.childCount === 0 && !"\r\n".includes(currentNode.type)) {
+            if (isLeafNodeWithCode(currentNode) || isHereDocBody(currentNode)) {
```

Comment:
I would prefer to pass that whole condition as first order function, just like `isComment`. Otherwise, we check for this isHereDocBody-Thing every iteration in every programming language and there are a lot of iterations for an usual syntax tree.

Reply:
Fixed